sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubBufferedProducerAsyncClient.java

```diff
@@ -0,0 +1,247 @@
+     * {@code {yournamespace}.servicebus.windows.net}.
+     *
+     * @return The fully qualified Event Hubs namespace that the connection is associated with
+     */
+    public String getFullyQualifiedNamespace() {
+        return client.getFullyQualifiedNamespace();
+    }
+
+    /**
+     * Gets the Event Hub name this client interacts with.
+     *
+     * @return The Event Hub name this client interacts with.
+     */
+    public String getEventHubName() {
+        return client.getEventHubName();
+    }
+
+    /**
+     * Retrieves information about an Event Hub, including the number of partitions present and their identifiers.
+     *
+     * @return The set of information for the Event Hub that this client is associated with.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<EventHubProperties> getEventHubProperties() {
+        return client.getProperties();
+    }
+
+    /**
+     * Retrieves the identifiers for the partitions of an Event Hub.
+     *
+     * @return A Flux of identifiers for the partitions of an Event Hub.
+     */
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public Flux<String> getPartitionIds() {
+        return client.getPartitionIds();
+    }
+
+    /**
+     * Retrieves information about a specific partition for an Event Hub, including elements that describe the available
+     * events in the partition event stream.
+     *
+     * @param partitionId The unique identifier of a partition associated with the Event Hub.
+     *
+     * @return The set of information for the requested partition under the Event Hub this client is associated with.
+     *
+     * @throws NullPointerException if {@code partitionId} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<PartitionProperties> getPartitionProperties(String partitionId) {
+        return client.getPartitionProperties(partitionId);
+    }
+
+    public int getBufferedEventCount() {
+        AtomicInteger total = new AtomicInteger();
+        partitionBatchMap.forEach((String key, ConcurrentLinkedDeque<EventDataBatch> item) -> {
+            total.addAndGet(item.size());
+        });
+        return total.get();
+    }
+
+    public int getBufferedEventCount(String partitionId) {
+        return partitionBatchMap.get(partitionId).size();
+    }
+
+    public Mono<Void> enqueueEvent(EventData eventData) {
+        return enqueueEvent(eventData, DEFAULT_SEND_OPTIONS);
+    }
+
+    public Mono<Void> enqueueEvent(EventData eventData, SendOptions options) {
+        Objects.requireNonNull(eventData, "'eventData' cannot be null.");
+        CreateBatchOptions createBatchOptions = new CreateBatchOptions().setPartitionId(options.getPartitionId()).setPartitionKey(options.getPartitionKey());
+        EventDataBatch eventDataBatch = this.producer.createBatch(createBatchOptions).block();
```

Comment:
async method cannot have `block()` in it.

Reply:
Another question is whether we really requires the async version of the buffered producer.

I would assume if I call `enqueueEvent`, it just do enqueue the event and return. I won't expect blocking operation even if this is in `EventHubBufferedProducerClient`.

Reply:
> Another question is whether we really requires the async version of the buffered producer.
> 
> I would assume if I call `enqueueEvent`, it just do enqueue the event and return. I won't expect blocking operation even if this is in `EventHubBufferedProducerClient` (at least I don't expect return after event sent).

It seems like that the design meeting didn't mention this point. Maybe they just want to align with the other clients.